packages/material-ui/src/Divider/Divider.js

```diff
@@ -71,13 +138,27 @@ const Divider = React.forwardRef(function Divider(props, ref) {
           [classes.flexItem]: flexItem,
           [classes.light]: light,
           [classes.vertical]: orientation === 'vertical',
+          [classes.text]: children,
```

Comment:
It feels a bit odd to me that we are silently treating the `children` as text only, by naming the class keys with the `text` prefix. Should we replace the `text` with `children`, so that at least is more obvious?

One more thought regarding the class keys. Just by seeing them is not obvious which ones go on the root, and which ones go on the span element. It would be better if we can change the classes on the root to start with `hasChildren` prefix. For example in the Autocomplete component, we have the `hasClearIcon` which I believe is something similar. Then in the span element, we can just use the `children` prefix.

@oliviertassinari thoughts on this?

Reply:
> is not obvious which ones go on the root

Agree. I believe we have tried to prefix the classes applied to nested elements, `textSpan` and `textSpanVertical` would turn into `spanText` and `spanTextVertical` under this previous model.

> bit odd to me that we are silently treating the children as text only

In the Skeleton we have:

```jsx
const hasChildren = Boolean(other.children);

      className={clsx(
        classes.root,
        classes[variant],
        {
          [classes[animation]]: animation !== false,
          [classes.withChildren]: hasChildren,
          [classes.fitContent]: hasChildren && !width,
          [classes.heightAuto]: hasChildren && !height,
        },
        themeVariantsClasses,
        className,
      )}
```

I'm not sure what would work best here.

Reply:
@oliviertassinari @mnajdova I have added `span` prefix to the children's classes. I am thinking of changing the text thing with the content or children for the whole PR because we can add chips or any button in the Divider. What do you guys think about it?

Reply:
Sounds much better in my opinion